indexes/utxoindex/src/stores/store_manager.rs

```diff
@@ -39,13 +39,14 @@ impl Store {
 
     pub fn update_utxo_state(
         &mut self,
-        to_add: Option<&UtxoSetByScriptPublicKey>,
-        to_remove: Option<&UtxoSetByScriptPublicKey>,
+        to_add: &UtxoSetByScriptPublicKey,
+        to_remove: &UtxoSetByScriptPublicKey,
         try_reset_on_err: bool,
     ) -> Result<(), StoreError> {
-        let res = match to_remove {
-            Some(utxo_collection) => self.utxos_by_script_public_key_store.remove_utxo_entries(utxo_collection),
-            None => Ok(()),
+        let mut res = Ok(());
+
+        if !to_remove.is_empty() {
```

Comment:
nit: I think it'll be more elegant to put this condition inside add/remove_utxo_entries